storage/src/checker.rs

```diff
@@ -87,17 +103,64 @@ impl<S: WritableStorage> NodeStore<Committed, S> {
         }
         Ok(())
     }
+
+    fn find_leaked_areas(
+        &self,
+        leaked_range: Range<LinearAddress>,
+    ) -> Result<Vec<(LinearAddress, AreaIndex)>, CheckerError> {
+        let mut leaked = Vec::new();
+        let mut current_addr = leaked_range.start;
+        loop {
+            let (area_size_index, area_size) = self.area_index_and_size(current_addr)?;
+            leaked.push((current_addr, area_size_index));
+
+            // advance to the next area
+            let Some(next_addr) = current_addr.checked_add(area_size) else {
```

Comment:
nit: I prefer ok_or_else here, to focus on the computation instead of the error:
```
     let next_addr = current_addr.checked_add(area_size).ok_or_else(||
        Err(CheckerError::AreaOutOfBounds {
                    start: current_addr,
                    size: area_size,
                    bounds: LinearAddress::new(NodeStoreHeader::SIZE))?;